src/finchlite/compile/bufferized_ndarray.py

```diff
@@ -228,6 +238,75 @@ def construct_from_numba(self, numba_buffer):
             )
         )
 
+    def serialize_to_numba(self, obj: BufferizedNDArray):
```

Comment:
Do we need a custom serializer if we have already removed ndim from the fields?

Reply:
I think we need. Without it we fallback to `_serialize_asm_struct_to_numba` for `serialize_to_numba(...: AssemblyStructFType)` call and `_serialize_asm_struct_to_numba` uses `fmt.struct_fields`.